site/content/blog/2020-05-04-pritunl.md

```diff
@@ -0,0 +1,175 @@
+The Network Load Balancer (NLB) which is in front of the instances which has 5 listeners as shown below. There are 3 listeners for VPN (on ports 1194, 1195, 1196) that can be used for the servers inside the Pritunl. Currently, we are using only the port 1194, so the rest ones are for future usage.
+
+{{< figure src="/blog/2020-05-04-pritunl/Pritunl_NetworkLB.png" alt="Pritunl Network Load Balancer">}}
+
+
+#### Pritunl - Configuration
+Pritunl is installed via the [userdata](https://github.com/mattermost/mattermost-cloud-monitoring/tree/master/terraform/aws/modules/pritunl/userdata.sh). As the instances do not store any configuration items except the MongoDB URI, each instance needs toconnect to the MongoDB as it gets all the configuration is needed and then the instance joins the Pritunl cluster.
+
+
+##### Initial setup
+As per [Pritunl documentation](https://docs.pritunl.com/docs/configuration-5#initial-setup):
+- SSH into one of the pritunl instances by using [Session-manager](https://console.aws.amazon.com/systems-manager/session-manager/sessions?region=us-east-1),
+- run: `sudo pritunl default-password` to get the default username and password,
+- go to your login page of Pritunl and use the credentials from previous step,
+- create a user `pritunl` and add a new password.
+
+##### Organization and server setup
+As per [Pritunl documentation](https://docs.pritunl.com/docs/connecting):
+- go into *Users* section and create a new organization, i.e. `devs_org`
+- go into *Servers* section and create a new server,
+  - Name: `devs_server`
+  - Port: `1194` or any other port that you have setup for VPN access, check [NLB](#network-load-balancer)
+  - DNS Server: `10.0.0.2, 8.8.8.8` where 10.0.0.2 is the DNS resolution of the subnet (2nd IP of each subnet is used for DNS resolution and our VPC where Pritunl is running has peering with all the other VPCs, so it is able to resolve all the names inside our network)
+  - save it
+- remove the `0.0.0.0/0` route from the `devs_server`
+- select the `Attach Organization` to attach `devs_org` with `devs_server`
+- select the `Attach host` to attach the 2 hosts(instances) into it
+- select `Add Route` (make sure at the bottom the server points to `devs_server`)
+  - need to add the DNS resolution as we have set it up above on the `DNS Server`:
+    - 10.0.0.2/32 DNS resolution
+  - need to add all the routes that this server will have access(i.e. other VPCs)
+    (add routes and each comment for clarity)
+    i.e:
+    - 10.8.0.0/16 prod VPC
+    - 10.16.0.0/16 staging VPC
+    - wherever else you want the Pritunl/VPN to have access
+
+- Start the server with `Start Server` button
+
+
+
+#### Enabling OneLogin
+To enable Onelogin, Pritunl Enterprise (with subscription) is needed. Otherwise the configuration is not appearing into the `Settings` on the top right. Then, you will need to setup a new app inside OneLogin(admin is needed) and you will need to paste that information inside Pritunl:
```

Comment:
```suggestion
To enable Onelogin, Pritunl Enterprise (with subscription) is needed, otherwise the configuration won't display in the **Settings**. Then, you will need to set up a new app inside OneLogin (*admin* access is needed) and you will need to paste that information inside Pritunl:
```